src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1449,25 +1604,44 @@ private bool LowerNullableInference(TypeSymbol source, TypeSymbol target, ref Ha
                 return false;
             }
 
-            LowerBoundInference(source.GetNullableUnderlyingType(), target.GetNullableUnderlyingType(), ref useSiteDiagnostics);
+            ExactInference(source.GetNullableUnderlyingType(), target.GetNullableUnderlyingType(), ref useSiteDiagnostics);
             return true;
         }
 
-        private bool ExactNullableInference(TypeSymbol source, TypeSymbol target, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        //PROTOTYPE(tuples): inference from tuple type is exact since tuples are non-variant.
```

Comment:
once we allow  tuple type conversions other than identity, "exact" will not be enough.